src/json-editor.component.ts

```diff
@@ -122,7 +122,19 @@ export class JsonEditorComponent extends AbstractTrackerComponent implements OnI
     return previews;
   }
 
+  deleteField(name: string) {
+    // remove it from the record
+    this._record = this._record.remove(name);
+    this.jsonStoreService.setJson(this._record);
+    // remove the key too, so that it will not be displayed as empty
+    this.keys.splice(this.keys.indexOf(name), 1);
+  }
+
   getPathForField(field: string): Array<any> {
     return [field];
   }
+
+  onFieldAdd(field: string) {
```

Comment:
Above the parameter is called `name` and here `field`, are they the same thing? Can you a docstring on top explaining how the parameter looks like?